python/cuspatial/cuspatial/core/geoseries.py

```diff
@@ -768,63 +775,174 @@ def contains_properly(self, other, align=True):
+
+        Parameters
+        ----------
+        other
+            a cuspatial.GeoSeries
+        align=True
+            to align the indices before computing .contains or not. If the
+            indices are not aligned, they will be compared based on their
+            implicit row order.
+
+        Examples
+        --------
+        Test if two points are equal:
+        >>> point = cuspatial.GeoSeries(
+            [Point(0, 0)],
             )
-            point_result = cudf.Series(
-                df_result["pip"], index=cudf.RangeIndex(0, len(df_result))
+        >>> point2 = cuspatial.GeoSeries(
+            [Point(0, 0)],
             )
-            point_result.name = None
-        return point_result
+        >>> print(point.equals(point2))
+        0    True
+        dtype: bool
+
+        Test if two polygons are equal:
+        >>> polygon = cuspatial.GeoSeries(
+            [
+                Polygon([[0, 0], [1, 0], [1, 1], [0, 0]]),
+            ]
+        )
+        >>> polygon2 = cuspatial.GeoSeries(
+            [
+                Polygon([[0, 0], [1, 0], [1, 1], [0, 0]]),
+            ]
+        )
+        >>> print(polygon.equals(polygon2))
+        0    True
+        dtype: bool
+
+        Returns
+        -------
+
+        result : cudf.Series
+            A Series of boolean values indicating whether each feature in A
+            is equal to the corresponding feature in B.
+        """
+        return EqualsBinpred(self, other, align)()
+
+    def covers(self, other, align=True):
+        """Compute if a GeoSeries of features A covers a second GeoSeries of
+        features B. A covers B if no points on B lie in the exterior of A.
+
+        Parameters
+        ----------
+        other
+            a cuspatial.GeoSeries
+        align=True
+            align the GeoSeries indexes before calling the binpred
+
+        Returns
+        -------
+        result : cudf.Series
+            A Series of boolean values indicating whether each feature in the
+            input GeoSeries covers the corresponding feature in the other
+            GeoSeries.
+        """
+        return CoversBinpred(self, other, align)()
+
+    def intersects(self, other, align=True):
+        """Compute the intersections of two GeoSeries.
+
+        Parameters
+        ----------
+        other
+            a cuspatial.GeoSeries
+        align=True
+            align the GeoSeries indexes before calling the binpred
+
+        Returns
+        -------
+        result : cudf.Series
+            A Series of boolean values indicating whether the geometries of
+            each row intersect.
+        """
+        if contains_only_points(self) and contains_only_points(other):
+            return cudf.Series(EqualsBinpred(self, other, align)())
+        else:
+            return IntersectsBinpred(self, other, align)()
+
+    def within(self, other, align=True):
+        """An object is said to be within other if at least one of its points
```

Comment:
At least one? Since no points are in the exterior of the other, isn't this equivalent to (and clearer as): 
```suggestion
        """An object is said to be within other all of its points
```

Reply:
For `contains_properly, you are correct.

Reply:
This text came from GeoPandas mostly. Saying if at least one is implying that points can be on the interior or the boundary of the object.

Reply:
Got it.